lib/msal-common/src/client/SilentFlowClient.ts

```diff
@@ -60,13 +60,23 @@ export class SilentFlowClient extends BaseClient {
         const environment = request.authority || this.authority.getPreferredCache();
         const authScheme = request.authenticationScheme || AuthenticationScheme.BEARER;
         const cacheRecord = this.cacheManager.readCacheRecord(request.account, this.config.authOptions.clientId, requestScopes, environment, authScheme);
-
+        
         if (request.forceRefresh || 
             !StringUtils.isEmptyObj(request.claims) || 
             !cacheRecord.accessToken || 
             TimeUtils.isTokenExpired(cacheRecord.accessToken.expiresOn, this.config.systemOptions.tokenRenewalOffsetSeconds) ||
             TimeUtils.wasClockTurnedBack(cacheRecord.accessToken.cachedAt) ||
             (cacheRecord.accessToken.refreshOn && TimeUtils.isTokenExpired(cacheRecord.accessToken.refreshOn, 0))) {
+
+            // Update server telemetry manager with the cache outcome
+            if (!cacheRecord.accessToken) {
```

Comment:
Instead of doing these checks again, can we break down the big if statement above? That would help ensure every condition is covered now and in the future.

Reply:
I have broken down this if statement, it's ready for review